### PR TITLE
feat: Add customProgramDuration field to Program model

### PR DESCRIPTION
**Pull request by gilsantosjulian**
### Description
This PR adds a Custom Program Duration filed which describes the months in which a Program can takes. This filed override the current value that is has been displayed in the frontend side (Prospectus)

See an example - Right side - 9 months:
[iux-information-systems](https://www.edx.org/micromasters/iux-information-systems?index=product&queryID=079a2394626ce063d46d4e7db87e5d8f&position=1&linked_from=autocomplete&c=autocomplete)

### Related Ticket
[WS-3462](https://2u-internal.atlassian.net/browse/WS-3462)

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @gilsantosjulian!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.


:warning: We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://openedx.org/cla)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/openedx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.     If you've signed an agreement in the past, you may need to re-sign.  See     [The New Home of the Open edX Codebase](https://open.edx.org/blog/the-new-home-of-the-open-edx-codebase/)     for details.

    
Once you've signed the CLA, please allow 1 business day for it to be processed.  After this time, you can re-run the     CLA check by editing the PR title.  If the problem persists, you can tag the `@openedx/cla-problems` team in a     comment on your PR for further assistance.

File: course_discovery/apps/api/serializers.py [hunk 1]
```diff
@@ -1863,7 +1863,7 @@ def prefetch_queryset(cls, partner, queryset=None):
     class Meta:
         model = Program
         fields = (
-            'uuid', 'title', 'subtitle', 'type', 'type_attrs', 'status', 'marketing_slug', 'marketing_url',
+            'uuid', 'title', 'duration_override', 'subtitle', 'type', 'type_attrs', 'status', 'marketing_slug', 'marketing_url',
```
Comment (Ali-D-Akbar) [inline course_discovery/apps/api/serializers.py]: Where is this `duration_override` coming from?

**Review by Ali-D-Akbar — COMMENTED** [review]
Can you update the model tests for degree too? You can find them here: https://github.com/openedx/course-discovery/blob/9f6df36ebb87bf37395732be71a0a16769710ff9/course_discovery/apps/course_metadata/tests/test_models.py#L2739

File: course_discovery/apps/api/serializers.py (hunk 1 above)
Comment (gilsantosjulian) [reply course_discovery/apps/api/serializers.py]: Ohh sorry,  was test value that I forgot to remove. I'll remove it

File: course_discovery/apps/course_metadata/models.py [hunk 2]
```diff
@@ -3066,6 +3066,10 @@ class Degree(Program):
         blank=True,
         default='',
     )
+    custom_program_duration = models.IntegerField(
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/models.py]: The field name should be changed. To clarify:
 - does this override the existing _course_run_weeks_to_complete property? If yes, the name can be program_duration_override. 
 - Is the field intended for both edX & 2U programs/degrees?

Comment (gilsantosjulian) [reply course_discovery/apps/course_metadata/models.py]: Thanks Dawoud for your comment is really helpful.

- Yes, it overrides the existing property, in that order, I will update the new name to `program_duration_override`
- It's really just intended for edX products

Finally, according to the specifications of our team, the field will be of type CharField, the idea is to insert something like: “6-9 months”

**Review by ansabgillani — APPROVED** [review]

**Comment by ansabgillani** [comment]
> Thanks for the pull request, @gilsantosjulian! Please note that it may take us up to several weeks or months to complete a review and merge your PR.
> 
> Feel free to add as much of the following information to the ticket as you can:
> 
> * supporting documentation
> * [Open edX discussion forum threads](https://discuss.openedx.org/)
> * timeline information ("this must be merged by XX date", and why that is)
> * partner information ("this is a course on edx.org")
> * any other information that can help Product understand the context for the PR
> 
> All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).
> 
> Please let us know once your PR is ready for our review and all tests are green.
> 
> ⚠️ We can't start reviewing your pull request until you've submitted a [signed contributor agreement](https://openedx.org/cla) or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/openedx/edx-platform/blob/master/CONTRIBUTING.rst) file for more information. If you've signed an agreement in the past, you may need to re-sign. See [The New Home of the Open edX Codebase](https://open.edx.org/blog/the-new-home-of-the-open-edx-codebase/) for details.
> 
> Once you've signed the CLA, please allow 1 business day for it to be processed. After this time, you can re-run the CLA check by editing the PR title. If the problem persists, you can tag the `@openedx/cla-problems` team in a comment on your PR for further assistance.

@gilsantosjulian Just FYI, you will need to sign this CLA for this commit to be merged.

**Comment by mphilbrick211** [comment]
Hi @azanbinzahid! Could you please take a look at this and allow tests to run?

**Comment by azanbinzahid** [comment]
> Hi @azanbinzahid! Could you please take a look at this and allow tests to run?

Hi @mphilbrick211, apologies for the inconvenience. Team, please take a look @ansabgillani @DawoudSheraz @Ali-D-Akbar

**Review by Ali-D-Akbar — APPROVED** [review]

**Pull request merged**